src/vs/workbench/contrib/audioCues/browser/audioCueContribution.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IWorkbenchContribution } from 'vs/workbench/common/contributions';
+import { IDebugService } from 'vs/workbench/contrib/debug/common/debug';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { Event } from 'vs/base/common/event';
+import { isCodeEditor } from 'vs/editor/browser/editorBrowser';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { raceTimeout } from 'vs/base/common/async';
+import { FileAccess } from 'vs/base/common/network';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+
+export class AudioCueContribution extends DisposableStore implements IWorkbenchContribution {
+	constructor(
+		@IDebugService readonly debugService: IDebugService,
+		@IEditorService readonly editorService: IEditorService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IAccessibilityService private readonly accessibilityService: IAccessibilityService,
+	) {
+		super();
+
+		this.add(Event.runAndSubscribeWithStore(editorService.onDidActiveEditorChange, (_, store) => {
+			let lastLineNumber = -1;
+
+			const activeTextEditorControl = editorService.activeTextEditorControl;
+			if (isCodeEditor(activeTextEditorControl)) {
+				store.add(
+					activeTextEditorControl.onDidChangeCursorPosition(() => {
+						const model = activeTextEditorControl.getModel();
+						if (!model) {
+							return;
+						}
+						const position = activeTextEditorControl.getPosition();
+						if (!position) {
+							return;
+						}
+						const lineNumber = position.lineNumber;
+						if (lineNumber === lastLineNumber) {
+							return;
+						}
+						lastLineNumber = lineNumber;
+
+						const uri = model.uri;
+
+						const breakpoints = debugService.getModel().getBreakpoints({ uri, lineNumber });
```

Comment:
I find it weird that audio cues depend on debug. I would have expected a service for playing audio cues and debug having a dependency to that, not the other way around?  Otherwise this should probably be renamed `debugAudioCues`.

Reply:
Yeah it is a bit weird, i agree. The thing is that we need a contribution which will react on line changes and based on line (if it has breakpoint, folded area, errors) play different sounds.
So we could duplicate the line listening logic across various areas, or we can have it in one place that check for any of those. I am not sure what is the best approach.

Reply:
Ah, I wasn't aware this is for more things than breakpoints. The PR title talks about "Play a sound whenever a breakpoint is hit"

Reply:
I can rename this to `debugAudioCues`.

Reply:
@bpasero not right now. But i know more things will come in the future. So if we name this `debugAudioCues`, then we might need `errorsAudioCues` and so on...

Reply:
If we envision more clients in the future based on active selection then it is fine to leave this as is.